server/src/main/java/org/elasticsearch/monitor/jvm/HotThreads.java

```diff
@@ -104,6 +105,33 @@ public static void logLocalHotThreads(Logger logger, Level level, String prefix,
         }
     }
 
+    /**
+     * Capture and log the current threads on the local node. Unlike hot threads this does not sample and captures current state only.
+     * Useful for capturing stack traces for unexpectedly-slow operations in production. The resulting message might be large, so it is
+     * split per thread and logged as multiple entries.
+     *
+     * @param logger        The logger to use for the logging
+     * @param level         The log level to use for the logging.
+     * @param prefix        The prefix to emit on each chunk of the logging.
+     */
+    public static void logLocalCurrentThreads(Logger logger, Level level, String prefix) {
+        if (logger.isEnabled(level) == false) {
+            return;
+        }
+
+        try (var writer = new StringWriter()) {
+            new HotThreads().busiestThreads(500).threadElementsSnapshotCount(1).detect(writer, () -> {
+                logger.log(level, "{}: {}", prefix, writer.toString());
+                writer.getBuffer().setLength(0);
+            });
```

Comment:
I wonder whether it is worthwhile to add similar counting and summary of total number of threads similar to how hot-threads are logged. We don't really need the counting to preserve orders for decompressing, but it seems useful to have a summary on how many threads are logged?

Reply:
We synced offline and agreed that search with prefix (likely unqiue) should be able to get us all the relevant entries without counting and summary. They might still be useful if somehow ingestion had problem to take in all data. But that should be a rare case and we can have follow-ups if we really need them.